plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -246,18 +243,14 @@ public boolean connect() throws OpenShiftException {
 	}
 
 	protected boolean authorize() {
-		client.setAuthorizationStrategy(createAuthorizationStrategy());
 		try {
-			IAuthorizationContext context = client.getContext(client.getBaseURL().toString());
-			if (context.isAuthorized()) {
-				String username = context.getUser().getName();
-				String token = context.getToken();
-				updateAuthorized(username, token);
+			IAuthorizationContext context = client.getAuthorizationContext();
+			if (!context.isAuthorized() 
+					&& credentialsPrompter != null
+					&& promptCredentialsEnabled){
+				credentialsPrompter.promptAndAuthenticate(this, null);
 			} else {
-				if (isEnablePromptCredentials()
-						&& credentialsPrompter != null) {
-					credentialsPrompter.promptAndAuthenticate(this, null);
-				}
+				updateCredentials(context);
```

Comment:
If context is not authorized and prompter is null (or not enabled), updateCredentials will be called anyway. Is it as it should be?